lib/ansible/modules/cloud/azure/azure_rm_image.py

```diff
@@ -0,0 +1,291 @@
+module: azure_rm_image
+options:
+  resource_group_name:
+    description:
+      - The resource group name to use or create to host the deployed template
+    required: true
+  name:
+    description:
+      - The name of the image being created.
+    required: True
+  vm_name:
+    description:
+      - The name of the VM within the resource group to capture. Required when state=present.
+    required: false
+  location:
+    description:
+      - The location where the image should be stored. Should be within the resource group.
+    default: The location of the resource group
+    required: false
+  state:
+    description:
+      - The desired state of the image
+    default: present
+    choices: [present, absent]
+    required: false
+  tags:
+    description:
+        - Tags to assign to the image.
+    required: false
+
+short_description: Capture Azure Virtual Machine Images
+version_added: "2.5"
+
+'''
+
+EXAMPLES = '''
+- name: Capture image
+  hosts: 127.0.0.1
+  connection: local
+
+  tasks:
+    - name: Capture image
+      azure_rm_image:
+        subscription_id: "{{ subscription_id }}"
+        resource_group_name: "{{ resource_group_name }}"
+        client_id: "{{ secrets.client_id }}"
+        tenant_id: "{{ tenant_id }}"
+        client_secret: "{{ secrets.client_secret }}"
+        vm_name: "{{ vm_name }}"
+        name: "{{ vm-name }}-image"
+        location: "{{ location }}"
+        state: present
+
+    - name: Delete image
+      azure_rm_image:
+        subscription_id: "{{ subscription_id }}"
+        resource_group_name: "{{ resource_group_name }}"
+        client_id: "{{ secrets.client_id }}"
+        tenant_id: "{{ tenant_id }}"
+        client_secret: "{{ secrets.client_secret }}"
+        name: "{{ vm-name }}-image"
+        location: "{{ location }}"
+        state: absent
+'''
+
+RETURN = '''
+location:
+    description: The location where the image is created
+    returned: on success
+    type: string
+name:
+    description: The name of the new image
+    returned: on success
+    type: string
+resource_group:
+    description: The resource group where the image is created
+    returned: on success
+    type: string
+status:
+    description: The result of the operation
+    returned: always
+    type: string
+'''
+
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.compute.models import Image
+    from azure.mgmt.compute.models import SubResource
+# from azure.mgmt.compute.models import VirtualMachineCaptureParameters
+except:
+    pass
+
+
+class AzureRMImage(AzureRMModuleBase):
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            vm_name=dict(
+                type='str',
+                required=False
```

Comment:
Same with the documentation, `required=False` is the default and is usually omitted from the arg spec.